gazebo_ros/src/node.cpp

```diff
@@ -27,9 +29,98 @@ Node::~Node()
+    _sdf = _sdf->GetElement("ros");
+  }
+
+  // Initialize arguments
+  std::string name = node_name;
+  std::string ns = "";
+  std::vector<std::string> arguments;
+  std::vector<rclcpp::Parameter> initial_parameters;
+
+  // Override name if tag is present
+  if (_sdf->HasElement("node_name")) {
+    name = _sdf->GetElement("node_name")->Get<std::string>();
+  }
+
+  // Set namespace if tag is present
+  if (_sdf->HasElement("namespace")) {
+    ns = _sdf->GetElement("namespace")->Get<std::string>();
+  }
+
+  // Get list of arguments from SDF
+  sdf::ElementPtr argument_sdf = _sdf->GetElement("argument");
+  while (argument_sdf) {
+    std::string argument = argument_sdf->Get<std::string>();
+    RCLCPP_WARN(internal_logger(), "Got argument %s", argument.c_str());
+    arguments.push_back(argument);
+    argument_sdf = argument_sdf->GetNextElement("argument");
+  }
+
+  // Convert each parameter tag to a ROS parameter
+  sdf::ElementPtr parameter_sdf = _sdf->GetElement("parameter");
+  while (parameter_sdf) {
+    auto param = sdf_to_ros_parameter(parameter_sdf);
+    if (rclcpp::ParameterType::PARAMETER_NOT_SET != param.get_type()) {
+      initial_parameters.push_back(param);
+    }
+    parameter_sdf = parameter_sdf->GetNextElement("parameter");
+  }
+
+  // Use default context
+  auto context = rclcpp::contexts::default_context::get_global_default_context();
+
+  // Create node with parsed arguments
+  return CreateWithArgs(
+    name, ns,
+    context,
+    arguments, initial_parameters);
+}
+
 Node::SharedPtr Node::Create(const std::string & node_name)
 {
-  return Create<const std::string &>(node_name);
+  return CreateWithArgs(node_name);
+}
+
+rclcpp::Parameter Node::sdf_to_ros_parameter(sdf::ElementPtr const & _sdf)
+{
+  if (!_sdf->HasAttribute("name")) {
+    RCLCPP_WARN(internal_logger(),
+      "Ignoring tag because it has no attribute 'name'. Tag: %s", _sdf->ToString("").c_str());
+    return rclcpp::Parameter();
+  }
+  if (!_sdf->HasAttribute("type")) {
+    RCLCPP_WARN(internal_logger(),
+      "Ignoring tag because it has no attribute 'name'. Tag: %s", _sdf->ToString("").c_str());
+    return rclcpp::Parameter();
+  }
+
+  std::string name = _sdf->Get<std::string>("name");
+  std::string type = _sdf->Get<std::string>("type");
+  sdf::ParamPtr value = _sdf->GetValue();
+
+  if ("int" == type) {
+    return rclcpp::Parameter(name, _sdf->Get<int>());
+  } else if ("double" == type || "float" == type) {
+    return rclcpp::Parameter(name, _sdf->Get<double>());
+  } else if ("bool" == type) {
+    return rclcpp::Parameter(name, _sdf->Get<bool>());
+  } else if ("string" == type) {
+    return rclcpp::Parameter(name, _sdf->Get<std::string>());
+  } else {
+    RCLCPP_WARN(internal_logger(),
+      "Ignoring tag because attribute 'type' is invalid. Tag: %s", _sdf->ToString("").c_str());
+    return rclcpp::Parameter();
+  }
+}
+
+rclcpp::Logger Node::internal_logger()
+{
+  return rclcpp::get_logger("gazebo_ros_node");
```

Comment:
This means that all messages get printed with `gazebo_ros_node` in front instead of the namespaced node name. This could make it more difficult to identify where the messages are coming from. What's the motivation for using this common logger?

Reply:
This is a ```private static``` method used internally for info/errors in Node::Create (such as when ROS is initialized, parameters cannot be parsed, etc.

Reply:
Ah gotcha :wink: Thanks!